### ISSUE TITLE
Previously filled out field set to empty doesn't save the empty value

### ISSUE THREAD

**Issue by robilim** [statamic/workshop#31]
**Describe the bug**
I tried to save an entry that used to have a field with a value. I now want to clear that value and save it blank, but it retains the original value.

**To Reproduce**
For example using the `workshop:entry:edit` tag, there is a textarea field with the name `description` with the value "Test" and now want it to be empty ie. "". Upon save, the description retained is still "Test".  There was no whitelist added so the assumption is that it should allow the field as it is part of the fieldset editing.

**Expected behavior**
The value should be the last submitted value.

**Environment details (please complete the following information):**
 - Workshop Version 1.1.3
 - Statamic Version 2.11.12
 - Fresh Install
 - OS: macOS 10.14.4
 - Browser: Chrome
 - Web Server: Valet
 - PHP Version: 7.2.18

**Additional context**
Looking at the point in which the field no longer exists in the `$this->fields` variable I found in the `initializeFields` function. The `array_filter` strips out any keys with blank values.

**WorkshopController.php**
```
private function initializeFields()
{
        $fields = $this->filter(Request::except(['_token']));
        
        // `array_filter` strips out any field with blank value.
        $this->fields = array_filter($fields);

        //ideally it just sets the fields based above and uses the whitelist to filter further
	$this->fields = $fields;
   }
```

**Comment by mar627** [comment]
Same experience here.

Although, not for items contained within an array. I have a little to-do list that adds and removes list items with input capabilities. The blank values are saved as number 3 indicates:

```md
title: 'the page'
slug: the-page
services:
  1: 'item one'
  2: 'item two'
  3: ''
  4: 'item three'
```

And this is the html within the workshop form:
```html
  <ul id="services">
    {{ services }}
    <li>
      <input value="{{value}}" name="services[{{index}}]" type="text">
      <span onclick="$(this).parent().remove()">delete</span>
    </li>
    {{ /services }}
  </ul>
```

**Comment by dannyuk1982** [comment]
Also having this problem, Gentlemen any chance this can be looked at?

On `WorkshopController` changing line 90 from:

`$this->fields = array_filter($fields);`

to
`$this->fields = $fields;`

Stops this behaviour, but no doubt has some unintended consequences otherwise?

It's pretty important to be able to set fields to blank or 0 though.

**Comment by danielfowler** [comment]
I'd throw in one more little tidbit: in @mar627 's example, deleting all his array items would mean the HTML on submit doesn't have any array fields (because he is using JS to remove the input completely from the code).

I would hope for a fix that does not overwrite a field simply because it is excluded from the code.

An appropriate fix would hopefully be that when `<input name="array[]" value="{{ value }}"/>` exists in the code and the input has no value (or when the user has cleared out the value), only then would that array be re-written on save.

**Comment by dannyuk1982** [comment]
I Agree @danielfowler - I think the workflow should be something like:

- Is the field not set/POSTed etc. at all = leave it alone
- is the fiels set/POSTed etc. as an empty array, empty string, or 0 = change it to that

At the moment we have the case where you cannot store 0, or a blank text field, which is bonkers.
